Documentation : comment mettre à jour une PR par rapport à dev

AmauryCarrade opened the pull request:
La question de « comment utiliser `rebase` » (ou son corollaire « ça veut dire quoi `rebase` ? ») revenant régulièrement, j'ai jugé bon d'ajouter un paragraphe dans la documentation expliquant de quoi il s'agit, ainsi que son usage.

J'ai au passage corrigé quelques typos et erreurs de mise en forme dans la documentation.

J'ai enfin retiré cette recommandation de la documentation, parce que l'on _squash_ généralement les PR, et qu'elle n'est donc plus pertinente.

> Essayez d'éviter les commits dits inutiles (``fix previous commit``, ...). Si vous en avez dans votre pull-request, un *squash* sera effectué lors du *merge*.

### Contrôle qualité

Regarder le _diff_ riche et vérifier que l'explication proposée est claire et juste.

AmauryCarrade committed:
Documentation de la mise à jour d'une PR

- Ajoute une section dans la documentation des contributions à ZdS expliquant
  comment mettre à jour la branche d'une PR par rapport à upstream/dev, en
  utilisant un rebase, et l'intérêt d'une telle pratique.

artragis commented on doc/source/contributing.rst:
```diff
@@ -43,23 +43,40 @@ Quelques bonnes pratiques
+
+.. note::
+
+	Pour pouvoir *rebaser*, il vous faut avoir défini une *remote* ``upstream`` pointant vers le dépôt principal de ZdS (la *remote* ``origin`` devant pointer vers votre fork), de la façon suivante : ``git remote add upstream https://github.com/zestedesavoir/zds-site.git``.
```
peut être ajouter que cette manipulation n'est à faire qu'une fois et après c'est bon pour tous les rebase qui seront faits.

artragis commented on doc/source/contributing.rst:
```diff
@@ -43,23 +43,40 @@ Quelques bonnes pratiques
+
+C'est une opération assez simple, et très utile pour s'assurer que votre modification s'intègre toujours bien avec la toute dernière version du site (si d'autres contributeurs ont changé quelque chose, typiquement). Elle a l'avantage, par rapport à une classique fusion (``git merge``), d'intégrer les commits naturellement dans l'historique, sans en ajouter de supplémentaire inutile.
+
+Pour *rebase* en ligne de commande, utilisez la commande ``git pull --rebase upstream dev`` (c'est tout). Si vous utilisez une interface graphique pour git, la marche à suivre dépendra de cette interface.
```
il faut ajouter un paragraphe qui explique qu'en cas de conflit, il faut : 

- régler le conflit
- faire un `add` sur les fichiers modifiés (en GUI ça ressemble souvent à "mark as resolved"
- faire un `git rebase --continue` (en GUI "continue rebase")

AmauryCarrade committed:
Remarques d'artragis

- Explications sur la façon de résoudre un conflit.
- Précision sur l'ajout unique de la remote “upstream”.

AmauryCarrade commented:
Suggestions pertinentes et prises en compte !
